PagingWithNetworkSample/app/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/ui/RedditActivity.kt

```diff
@@ -80,33 +83,28 @@ class RedditActivity : AppCompatActivity() {
 
     private fun initAdapter() {
         val glide = GlideApp.with(this)
-        val adapter = PostsAdapter(glide) {
-            model.retry()
+        adapter = PostsAdapter(glide)
+        list.adapter = adapter.withLoadStateHeaderAndFooter(
+            header = PostsLoadStateAdapter(adapter),
+            footer = PostsLoadStateAdapter(adapter)
+        )
+
+        lifecycleScope.launch {
+            adapter.loadStateFlow.collect { loadStates ->
```

Comment:
collectLatest

(not important here, just good habit)

Reply:
Done